resources/js/components/Product/Price.vue

```diff
@@ -0,0 +1,130 @@
+<script>
+export default {
+    props: {
+        product: {
+            type: Object,
+            default: () => config.product || {},
+        },
+        type: {
+            type: String,
+            default: 'catalog',
+        },
+        options: {
+            type: Object,
+            default: {},
+        },
+    },
+    render() {
+        return this.$scopedSlots.default(Object.assign(this, { self: this }))
+    },
+
+    data: () => ({
+        mounted: false,
+    }),
+
+    mounted() {
+        this.mounted = true
+    },
+
+    methods: {
+        getTaxPercent(product) {
+            let taxClass = product.tax_class_id ?? product
+            let taxValues = product.tax_values ?? window.config.tax.values[taxClass] ?? {}
+
+            let groupId = this.$root.user?.group_id ?? 0 // 0 is always the NOT_LOGGED_IN group
+            let customerTaxClass = window.config.tax.groups[groupId] ?? 0
+
+            return taxValues[customerTaxClass] ?? 0
+        },
+
+        calculatePrice(product, location, options = {}) {
+            let special_price = options.special_price ?? false
+            let displayTax = this.$root.includeTaxAt(location);
+
+            let price = special_price
+                ? product.special_price ?? product.price ?? 0
+                : product.price ?? 0
+
+            if(options.product_options) {
+                price += this.calculateOptionsValue(price, product, options.product_options)
+            }
+
+            let taxMultiplier = this.getTaxPercent(product) + 1
+
+            if (window.config.tax.calculation.price_includes_tax == displayTax) {
+                return price
+            }
+
+            return displayTax
+                ? price * taxMultiplier
+                : price / taxMultiplier
+        },
+
+        calculateOptionsValue(basePrice, product, customOptions) {
+            let addition = 0
+
+            Object.entries(customOptions).forEach(([key, val]) => {
+                if (!val) {
+                    return
+                }
+
+                let option = product.options.find((option) => option.option_id == key)
+                let optionPrice = ['drop_down', 'radio'].includes(option.type)
+                    ? option.values.find((value) => value.option_type_id == val).price
+                    : option.price
+
+                if (optionPrice.price_type == 'fixed') {
+                    addition += parseFloat(optionPrice.price)
+                } else {
+                    addition += (parseFloat(basePrice) * parseFloat(optionPrice.price)) / 100
+                }
+            })
+
+            return addition
+        }
+    },
+
+    computed: {
+        specialPrice() {
+            if(!this.mounted) {
+                return;
+            }
+            return this.calculatePrice(this.product, this.type, Object.assign(this.options, { special_price: true }))
```

Comment:
Is this what Magento defines as the special price, or the final price?

https://magento.stackexchange.com/questions/279738/what-is-difference-between-price-special-price-and-final-price-in-magento

I think final price may be more descriptive of what you're actually getting here.